### DISCUSSION TITLE
EP600 ac_output_power

### DISCUSSION THREAD

**Discussion by KM011092** [Patrick762/hassio-bluetti-bt#91] [General]
Hi, I’ve decided to start using Bluetti-BT instead of my own fork of the original Bluetti-MQTT code to monitor my EP600. I’ve managed to get everything up and running, but having trouble to understand the ac_output_power values. Can someone help me to understand what these are ? They don’t seem to match with anything I can see in the Bluetti App.
I am also missing the values the Bluetti App shows under “Consumption”, would someone be able to add them to the Bluetti-BT code if I provide the Modbus addresses ?

**Comment by Patrick762** [comment]
I think I already used some of your values from the bluetti_mqtt Pull Request (I think the PR was from you)

**Comment by Patrick762** [comment]
The latest version is in the dev branch https://github.com/Patrick762/hassio-bluetti-bt/blob/dev/custom_components%2Fbluetti_bt%2Fbluetti_bt_lib%2Fdevices%2Fep600.py
Including a new structure to manage the data polling

**Comment by Patrick762** [comment]
You can also join the new discord server https://discord.com/invite/FtArp8j2QZ

**Reply by KM011092** [reply]
Can anyone see my posts on the discord server, or am I doing something wrong ?

**Reply by Patrick762** [reply]
Yes, I can see them. I just didn't have the time to look at the issue yet. I might have some spare time on friday afternoon

**Reply by KM011092** [reply]
Thanks for the confirmation. No urgency, please take your time and look at it whenever it works for you.

**Comment by KM011092** [comment]
I am using the bluetti-bt integration for a while now to monitor my EP600 and I really like it. @Patrick762: Thanks a lot for the great work and sharing this with us !!!

There are a few things I discovered / I don't understand and I also wanted to contribute by sharing some information I have.

1. AC_Output_* values. I am watching these values since a while, but don't understand what they are showing. I've also searched the Bluetti app and could not find anything matching there. I assume that AC_Output is supposed to show the values for "Consumption / Verbrauch" in the Bluetti App. If that assumption is wrong, I would really like to understand what these values are about and it would be great if you could add the values showing under "Consumption / Verbrauch" to your integration.
If theAC_Output_* should show the consumption values, I do think the integration is pulling incorrect modbus addresses. 

Here are the Modbus values I was using when I still ran bluetti-MQTT: 
self.struct.add_decimal_field("ac_output_frequency", ?????,   1) - did not have the frequency
self.struct.add_int_field("ac_output_power_phase1", 1430)
self.struct.add_int_field("ac_output_power_phase2", 1436)
self.struct.add_int_field("ac_output_power_phase3", 1442)
self.struct.add_decimal_field("ac_output_voltage_phase1",   1431, 1)
self.struct.add_decimal_field("ac_output_voltage_phase2",   1437, 1)
self.struct.add_decimal_field("ac_output_voltage_phase3",   1443, 1)
self.struct.add_decimal_field("ac_output_current_phase1",   1432, 1)
self.struct.add_decimal_field("ac_output_current_phase2",   1438, 1)
self.struct.add_decimal_field("ac_output_current_phase3",   1444, 1)

3. I am missing the summarized values for Production/Erzeugung, Grid/Stromnetz and Consumption/Verbrauch shown on the main screen in the Bluetti app

Here are the modbus addresses from bluetti-mqtt
self.struct.add_uint_field('pv_input_power_all', 144)  # Total PV in Production/Erzeugung
self.struct.add_uint_field('ac_output_power', 142)  # Total AC out Consumption/Verbrauch
self.struct.add_uint_field('grid_power', 146)  # Total Grid in - value only +/- unknown Grid/Strom Netz
self.struct.add_uint_field('total_battery_percent', 102) - bluetti-bt is discovering this value, but I could not find how, as it's not listed  in EP600.py

Thanks again for the bluetti-bt integration.

**Reply by Patrick762** [reply]
Thanks for the details. I'll look into it as soon as I have time (still too much other projects). If you want you can always try to improve the integration and create a pull request

**Reply by KM011092** [reply]
No worries, it's not urgent. If I find the time I may start working on a pull request. Nevertheless since I am not a developer, that would only be copy and paste / try and error :-).
Can you give me a hint how I can get a forked version with changes installed in home assistant ? I was not able to work that out yet. Is it just downloading the code from github and replace the files in the custom_components folder ?

**Reply by Patrick762** [reply]
In theory it should work like that. But you have to restart after that.

**Reply by KM011092** [reply]
FYI - I started work to add above values and will submit a pull request, once done.
